engine/select_optimal_members.py

```diff
@@ -0,0 +1,305 @@
+import os
+from collections import Counter
+from datetime import datetime
+
+import click
+import numpy as np
+
+from engine.check import check_intersection, check_variable
+from engine.tolerance import tolerance
+from util.click_util import cli_help
+from util.dataframe_ops import compute_rel_diff_dataframe, parse_probtest_csv
+from util.log_handler import logger
+
+
+def select_members(
+    stats_file_name,
+    member_type,
+    min_member_num,
+    max_member_num,
+    total_member_num,
+    min_factor,
+    max_factor,
+):
+
+    members = [i for i in range(1, total_member_num + 1)]
```

Comment:
Hmm, I think it's nicer to have the members starting at 1. Also this would mean a lot of changes, including updating all optimal member selections in ICON in [set_probtest_ensemble_num](run/tolerance/set_probtest_ensemble_num), as they are based on the members starting at 1...

Reply:
Again, I think using starting index 1 in a zero based language makes the code more complicated as needed. But for the members I leave it up to you.

Reply:
Then I decide for the members to leave it like this ;-)